go-controller/pkg/cni/cni.go

```diff
@@ -395,6 +395,19 @@ func getCNIResult(pr *PodRequest, getter PodInfoGetter, podInterfaceInfo *PodInt
 		ips = append(ips, ip)
 	}
 
+	// mimic an ip entry with gateway in order to trick multus to set this interface as default: true
+	ramsamsamMAC, _ := net.ParseMAC("00:AA:BB:CC:DD:EE")
+	if podInterfaceInfo.MAC.String() == ramsamsamMAC.String() {
+		ips = append(ips, &current.IPConfig{
+			Interface: current.Int(1),
+			Gateway:   podInterfaceInfo.Gateways[0],
+			Address: net.IPNet{
+				IP:   net.ParseIP("0.0.0.0"),
+				Mask: net.CIDRMask(0, 32),
+			},
```

Comment:
OK, I get what you're doing. 

Can you please try to close the loop in this PoC by providing a custom unmarshalling fn to the OVN-K cni `Response` object that allows you to not have to put this address in the CNI response ? 
@RamLavi

Reply:
NP, I [introduced](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5226/files#diff-29e92a5d3cc7040dfb9574fb946356e55fa5161996ac792558468b550e23048cR113) a new Unmarshal method for the Response object, that will safely unmarshal it even if the `Address` is empty.
It is [POCed](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5226/files#diff-07da4a6ce27a5365146fbebc7f946d2eb6ccc0cb8e89d066c01259da752d3377R279) in the unit test.

However note that using this new Unmarshal method will mean changing multus-cni code in order to work, which means another set of approvers to convince. So not a big issue, just an FYI consideration.